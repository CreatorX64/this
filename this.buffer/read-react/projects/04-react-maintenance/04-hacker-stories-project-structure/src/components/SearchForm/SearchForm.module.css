/* If using CSS-in-JS, where no CSS file is needed, one could still have a
separate style.js file for all the styled components */

.searchForm {
  display: flex;
  align-items: baseline;
  gap: 8px;
  padding: 10px 0 20px;
}
