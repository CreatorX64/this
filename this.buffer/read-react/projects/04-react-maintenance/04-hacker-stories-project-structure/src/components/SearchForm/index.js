import InputWithLabel from "../InputWithLabel";
import styles from "./SearchForm.module.css";

const SearchForm = ({ searchTerm, onSearchInput, onSearchSubmit }) => {
  return (
    <form onSubmit={onSearchSubmit} className={styles.searchForm}>
      <InputWithLabel
        id="search"
        value={searchTerm}
        onInputChange={onSearchInput}
        isFocused
      >
        <strong>Search:</strong>
      </InputWithLabel>

      <button
        type="submit"
        disabled={!searchTerm || !searchTerm.trim()}
        className="button button-large"
      >
        Submit
      </button>
    </form>
  );
};

export default SearchForm;
