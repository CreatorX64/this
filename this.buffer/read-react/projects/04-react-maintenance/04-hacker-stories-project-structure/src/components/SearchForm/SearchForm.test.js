import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import SearchForm from "./index";

describe("SearchForm", () => {
  const searchFormProps = {
    searchTerm: "React",
    onSearchInput: jest.fn(),
    onSearchSubmit: jest.fn()
  };

  test("renders the input field with its value", () => {
    render(<SearchForm {...searchFormProps} />);

    expect(screen.getByDisplayValue("React")).toBeInTheDocument();
  });

  test("renders the correct label", () => {
    render(<SearchForm {...searchFormProps} />);

    // getByLabelText search function allows us to find an element by a label
    // in a form. This is useful for components that render multiple labels
    // and HTML controls
    expect(screen.getByLabelText(/Search/)).toBeInTheDocument();
  });

  test("calls onSearchInput on input field change", () => {
    render(<SearchForm {...searchFormProps} />);

    fireEvent.change(screen.getByDisplayValue("React"), {
      target: { value: "Redux" }
    });

    expect(searchFormProps.onSearchInput).toHaveBeenCalledTimes(1);
  });

  test("calls onSearchSubmit on button submit click", () => {
    render(<SearchForm {...searchFormProps} />);

    fireEvent.submit(screen.getByRole("button"));

    expect(searchFormProps.onSearchSubmit).toHaveBeenCalledTimes(1);
  });

  // Snapshot testing: Facebook created snapshot tests as a more lightweight way
  // to test React components and their structure. A snapshot test creates an
  // instance of your rendered componentâ€™s output as HTML elements and their structure.
  // This snapshot is compared to the same snapshot in the next test to give more
  // output on how the rendered component changed and show why any tests failed in
  // the difference. You can accept or deny any differences in your source code until
  // the component functions as intended
  test("renders snapshot", () => {
    const { container } = render(<SearchForm {...searchFormProps} />);
    expect(container).toMatchSnapshot();
  });
});
