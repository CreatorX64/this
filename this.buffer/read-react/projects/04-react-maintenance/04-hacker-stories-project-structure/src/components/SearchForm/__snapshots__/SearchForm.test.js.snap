// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm renders snapshot 1`] = `
<div>
  <form
    class="searchForm"
  >
    <label
      class="label"
      for="search"
    >
      <strong>
        Search:
      </strong>
    </label>
    Â 
    <input
      class="input"
      id="search"
      type="text"
      value="React"
    />
    <button
      class="button button-large"
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
